src/main/java/org/skriptlang/skriptworldguard/elements/expressions/ExprRegionFlag.java

```diff
@@ -0,0 +1,118 @@
+package org.skriptlang.skriptworldguard.elements.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import com.sk89q.worldguard.WorldGuard;
+import com.sk89q.worldguard.protection.flags.*;
+import com.sk89q.worldguard.protection.regions.ProtectedRegion;
+import org.bukkit.event.Event;
+import org.skriptlang.skriptworldguard.SkriptWorldGuard;
+import org.skriptlang.skriptworldguard.worldguard.WorldGuardRegion;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExprRegionFlag extends PropertyExpression<WorldGuardRegion, Flag> {
+
+    static {
+        Skript.registerExpression(ExprRegionFlag.class, Flag.class, ExpressionType.COMBINED,"[worldguard] flag[s] %strings%", "worldguardregions");
+    }
+
+    WorldGuard wg = WorldGuard.getInstance();
+    private Expression<String> exprFlag;
+    private Expression<WorldGuardRegion[]> exprRegions;
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public boolean init(Expression<?>[] expression, int matchedPattern, Kleenean kleenean, ParseResult parseResult) {
+        exprFlag = (Expression<String>) expression[0];
+        exprRegions = (Expression<WorldGuardRegion[]>) expression[1];
+        return true;
+    }
+
+    @Override
+    protected Flag[] get(Event event, WorldGuardRegion[] region) {
+        Flag<?> flag = Flags.fuzzyMatchFlag(wg.getFlagRegistry(), exprFlag.getSingle(event));
+        List<Flag> result = new ArrayList<>();
+        for (WorldGuardRegion wrg : region){
+            ProtectedRegion rg = wrg.getRegion();
+            if (rg != null){
+                result.add((Flag) rg.getFlag(flag));
+            }
+        }
+        return (Flag[]) result.toArray();

```

Comment:
```suggestion
        return result.toArray(new Flag[0]);
```
this is likely cleaner